### ISSUE TITLE
Bundled output shouldn't include babel-polyfill

### ISSUE THREAD

**Issue by ColinEberhardt** [trusktr/infamous#114]
Hi - just had someone raise an issue when trying to combine our libraries:

https://github.com/d3fc/d3fc/issues/1068

Looks like we both include babel-polyfill in our library bundles, which causes integration issues. I don't think either should include the polyfill, leaving it up to the consumer.

**Comment by trusktr** [comment]
Hello Colin, sorry for the late reply, I just noticed this issue. Thanks for reaching out!

I wonder what we can do about this. The reason I had included it in the first place was so that it was 'easy' for someone to install and simply get going. I didn't know this would be a problem until now; I hadn't ran into this issue before.

Hmmm, asking end users to manually include babel-polyfill in their apps is a type of thing I wanted to avoid. Let me think about this in the next days and get back to you. :}

**Comment by ColinEberhardt** [comment]
No worries. We removed the polyfill from d3fc.

As a general principle I think it is up the creator of a library to state what browser versions (or better still what browser features) they require.

It is up to the consumer to determine whether this is compatible with the browsers they wish to target for their application, and polfyill accordingly.

Basically, as library authors, I don't think we should be making that call!

**Comment by trusktr** [comment]
It's an interesting topic, because doing that does raise the barrier of entry. For example, suppose I just want new programmers to follow a set of my (not-yet-existent) tutorials, and for it to just be as easy as place the script tag, then write this short HTML and JS, and voila.

I wonder if instead of throwing an error, if it would be better practice for babel-polyfill to send a warning to console, but otherwise prefer the first implementation that is found and just allow to continue.

**Comment by trusktr** [comment]
I am now building infamous with Buble for CommonJS publication on NPM, and Webpack+Buble for the global bundle (@jibesh was having the issue with the global bundle). The code is much leaner and smaller now! Woot woot!

I'm swapping Webpack for Rollup, which will make it even more lean (WIP)!

**Comment by trusktr** [comment]
For some reason, when using Buble, this bug with Two.js presented itself in the build: https://github.com/jonobr1/two.js/issues/241, which has led me to [create a `postinstall` script](https://github.com/trusktr/infamous/commit/729fe3c09005278148a1d6a59835b6e038d1353c#diff-adccad3eefeff7a45feabbde59f94d92R1) that modifies Two.js' built file after install. However, this fix is highly app-specific and not good for a library, because the script will fail in apps using different versions of NPM where the location of `two.js` might be different.

**Comment by trusktr** [comment]
Ah, well my PR on Two.js was merged, so now it won't be a problem. https://github.com/jonobr1/two.js/pull/242

**Comment by trusktr** [comment]
Oh geez. I could've just linked my fork of two.js in package.json, and not have written that script at all. 😆

**Comment by trusktr** [comment]
Going to test in supported browsers before releasing this. tracking in https://github.com/trusktr/infamous/issues/119

**Comment by trusktr** [comment]
Released in [v17.0.5](https://github.com/trusktr/infamous/releases/tag/v17.0.5)

**Comment by trusktr** [comment]
I'm back to using Babel, but with `transform-runtime`'s `polyfill: true` option which doesn't pollute global scope like babel-polyfill does, so code has output like

```js
var _symbol = require('babel-runtime/core-js/symbol');

var _symbol2 = _interopRequireDefault(_symbol);
```

I think this is a better approach, and achieves basically the same thing. I've got everything running in IE 11 now (that's the lowest target for infamous).
